spec/controllers/users_controller_spec.rb

```diff
@@ -95,9 +97,69 @@
     end
   end
 
+  context "GET confirm_email" do
+    let(:unconfirmed_user) { FactoryGirl.create(:unconfirmed_user)}
+    let(:another_activation_token) { SecureRandom.hex(8) }
+
+    context "if user not signed in: " do
+      context "if user not activated: " do
+        it "gets valid activation link and redirects to new_user_sessions_path" do
+          get :confirm_email, { email: unconfirmed_user.email, token: unconfirmed_user.activation_token }
+          expect(assigns(:user)).to eq(unconfirmed_user)
+          expect(response).to redirect_to(new_user_sessions_path)
+        end
+
+        it "gets invalid token and redirects to new_user_session_path" do
+          get :confirm_email, { email: unconfirmed_user.email, token: unconfirmed_user.activation_token }
+          expect(assigns(:user)).to eq(unconfirmed_user)
+          expect(response).to redirect_to(new_user_sessions_path)
+        end
+
+        it "gets no token and redirects to new_user_session_path" do
```

Comment:
It might be little easier to understand why in all cases user is redirected to `new_user_session_path` if flash message was tested - currently it looks like no matter what you provide in params it will always do the same :/